include/odp/api/spec/traffic_mngr.h

```diff
@@ -583,6 +614,22 @@ void odp_tm_egress_init(odp_tm_egress_t *egress);
 int odp_tm_capabilities(odp_tm_capabilities_t capabilities[],
 			uint32_t              capabilities_size);
 
+/** Query TM Capabilities specific to an egress
+ *
+ * The function returns the set of TM limits supported by this implementation
+ * for a given egress. Unlike odp_tm_capability() which return's capabilities
+ * of already created TM system which are limited by its requirements, this
+ * function returns maximum TM system limits.
+ *
+ * @param[out] capabilities      odp_tm_capabilities_t record to be filled in.
+ * @param      egress            Only capabilities compatible with this egress
+ *                               are returned.
+ * @retval 0  on success
+ * @retval <0 on failure
+ */
+int odp_tm_egress_capabilities(odp_tm_capabilities_t *capabilities,
+			       odp_tm_egress_t       *egress);
```

Comment:
In ODP it is customary that in case that a feature is not supported, the corresponding capability function does not return a failure but successfully returns a capability structure that indicates that the feature is not supported. Failure of a capability function should not normally happen and indicates some sort of internal error. When reading the validation test I realized that you have a different idea of how this function works.

Another issue is that it may be unclear how this function behaves when called with a pktio that has not been opened in the TM output mode. 

To clarify the semantics, maybe the following should be added in the description:

```
Lack of TM support in the given egress does not cause this
function to return a failure. Lack of TM support is indicated
by zero max_tm_queues capability.

If the pktio of an egress of the pktio kind has not been opened
in the ODP_PKTOUT_MODE_TM pktout mode, the capabilities will
indicate that TM is not supported.
```

Another possibility would be to return a failure if TM is not supported, but then maybe the failure should be distinguished from other kind of failures by having different return values. But it seems that typically capability functions in ODP do not fail.

Reply:
Ack, I forgot about that. Will fix the usage in test and comment.